### ISSUE TITLE
autoreconf: automake failed with exit status: 1

### ISSUE THREAD

**Issue by SamuelMarks** [deric/mesos-deb-packaging#19]
Ubuntu 14.04 x64. Running `./build_mesos --ref 0.27.0 --build-version p1` on Ubuntu 15.10 x64 works fine.

```
~/mesos-deb-packaging$ ./build_mesos --ref 0.27.0 --build-version p1
Trying /etc/os-release...
Found directory `mesos-repo'; skipping checkout.
git checkout -f 0.27.0
HEAD is now at c29b3b6... Fixed missing include for V1 `Resources.cpp`.
ls: cannot access /home/ubuntu/mesos-deb-packaging/*.egg: No such file or directory
configure.ac:1474: warning: cannot check for file existence when cross compiling
../../lib/autoconf/general.m4:2777: AC_CHECK_FILE is expanded from...
configure.ac:1474: the top level
3rdparty/Makefile.am:107: error: Libtool library used but 'LIBTOOL' is undefined
3rdparty/Makefile.am:107:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
3rdparty/Makefile.am:107:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
3rdparty/Makefile.am:107:   If 'LT_INIT' is in 'configure.ac', make sure
3rdparty/Makefile.am:107:   its definition is in aclocal's search path.
3rdparty/Makefile.am:84: warning: source file '$(RY_HTTP_PARSER)/http_parser.c' is in a subdirectory,
3rdparty/Makefile.am:84: but option 'subdir-objects' is disabled
automake: warning: possible forward-incompatibility.
automake: At least a source file is in a subdirectory, but the 'subdir-objects'
automake: automake option hasn't been enabled.  For now, the corresponding output
automake: object file(s) will be placed in the top-level directory.  However,
automake: this behaviour will change in future Automake versions: they will
automake: unconditionally cause object files to be placed in the same subdirectory
automake: of the corresponding sources.
automake: You are advised to start using 'subdir-objects' option throughout your
automake: project, to avoid future incompatibilities.
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/adaptor_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/base64_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/bits_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/bytes_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/cache_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/duration_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/dynamiclibrary_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/error_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/flags_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/gzip_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/hashmap_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/hashset_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/interval_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/ip_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/json_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/jsonify_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/linkedhashmap_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/mac_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/main.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/multimap_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/none_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/numify_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/option_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/os_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/os/sendfile_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/os/signals_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/os/strerror_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/path_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/protobuf_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/protobuf_tests.pb.cc' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/recordio_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/result_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/set_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/some_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/strings_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/subcommand_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/svn_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/try_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/uuid_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:160: warning: source file '$(STOUT)/tests/version_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:160: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:205: warning: source file '$(STOUT)/tests/proc_tests.cpp' is in a subdirectory,
3rdparty/Makefile.am:205: but option 'subdir-objects' is disabled
3rdparty/Makefile.am:127: warning: variable 'GLOG_LDFLAGS' is defined but no program or
3rdparty/Makefile.am:127: library has 'GLOG' as canonical name (possible typo)
Makefile.am:43: error: Libtool library used but 'LIBTOOL' is undefined
Makefile.am:43:   The usual way to define 'LIBTOOL' is to add 'LT_INIT'
Makefile.am:43:   to 'configure.ac' and run 'aclocal' and 'autoconf' again.
Makefile.am:43:   If 'LT_INIT' is in 'configure.ac', make sure
Makefile.am:43:   its definition is in aclocal's search path.
Makefile.am:45: warning: source file 'src/authenticator_manager.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/authenticator.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/clock.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/firewall.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/help.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/http.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/io.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/latch.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/logging.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/metrics/metrics.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/pid.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/poll_socket.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/profiler.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/process.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/reap.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/socket.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/subprocess.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/time.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:45: warning: source file 'src/timeseries.cpp' is in a subdirectory,
Makefile.am:45: but option 'subdir-objects' is disabled
Makefile.am:84: warning: source file 'src/libevent_ssl_socket.cpp' is in a subdirectory,
Makefile.am:84: but option 'subdir-objects' is disabled
Makefile.am:84: warning: source file 'src/openssl.cpp' is in a subdirectory,
Makefile.am:84: but option 'subdir-objects' is disabled
Makefile.am:84: warning: source file 'src/ssl/utilities.cpp' is in a subdirectory,
Makefile.am:84: but option 'subdir-objects' is disabled
Makefile.am:108: warning: source file 'src/libevent.cpp' is in a subdirectory,
Makefile.am:108: but option 'subdir-objects' is disabled
Makefile.am:108: warning: source file 'src/libevent_poll.cpp' is in a subdirectory,
Makefile.am:108: but option 'subdir-objects' is disabled
Makefile.am:113: warning: source file 'src/libev.cpp' is in a subdirectory,
Makefile.am:113: but option 'subdir-objects' is disabled
Makefile.am:113: warning: source file 'src/libev_poll.cpp' is in a subdirectory,
Makefile.am:113: but option 'subdir-objects' is disabled
Makefile.am:194: warning: source file 'src/tests/benchmarks.cpp' is in a subdirectory,
Makefile.am:194: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/collect_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/decoder_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/encoder_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/future_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/http_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/io_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/limiter_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/main.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/metrics_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/mutex_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/owned_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/process_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/queue_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/reap_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/sequence_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/shared_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/statistics_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/subprocess_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/system_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/timeseries_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:148: warning: source file 'src/tests/time_tests.cpp' is in a subdirectory,
Makefile.am:148: but option 'subdir-objects' is disabled
Makefile.am:190: warning: source file 'src/tests/ssl_tests.cpp' is in a subdirectory,
Makefile.am:190: but option 'subdir-objects' is disabled
Makefile.am:186: warning: source file 'src/tests/ssl_client.cpp' is in a subdirectory,
Makefile.am:186: but option 'subdir-objects' is disabled
autoreconf: automake failed with exit status: 1
```

**Comment by deric** [comment]
Looks like `libtool` is missing, try installing `apt-get install libtool-bin`.

**Comment by SamuelMarks** [comment]
Thanks, a `apt-get install libtool` did the trick, was building [tar] from source then got a weird error near the end, checked Google that told brought me to some Redis build log with the word "libtool", and I tried installing it and that did the trick.

Reading your reply made me retry mesos-deb-packaging, which also worked.

:)

PS: Not sure how I missed the dependency (which is clearly in your README!)
PPS: Building from tar is taking a very long time, I ran it with `time` and will post the duration. More than twice as long as building this deb. I'm guessing you're not building the entire thing, and only selected components?
